launch_testing/launch_testing/__init__.py

```diff
@@ -53,38 +57,57 @@ def __init__(
         self.name = name
         self.launch_descriptor = launch_descriptor
         self.expected_lines = expected_lines
-        self.expected_output = b'\n'.join(self.expected_lines)
+        if regex_match:
+            self.expected_output = self.expected_lines
+            # Add a surrounding capture group
+            self.expected_output_captured = \
+                b'(?P<launch_testing_capture>' + self.expected_output + b')'
+            # Increment any group IDs in the original regex to compensate for the surrounding group
+            self.expected_output_captured = re.sub(
+                b'\\\(\d)',  # TODO(dhood): only match single backslashes
+                lambda matchobj: b'\\' + str(int(matchobj.group(1)) + 1).encode(),
+                self.expected_output_captured)
```

Comment:
open to suggestions of less intrusive approaches to this. 
the situation is that I want to check if the lines received are _completely_ covered by the regex. I wrap the regex in a capture group and check that the captured group contains the entirety of the text received. because of that, I then have to increment any references to groups inside the origin regex because `\1` then refers to the outer group, not whatever group it originally referred to. (which is a problem for something like [this](https://github.com/ros2/demos/blob/master/intra_process_demo/test/cyclic_pipeline.regex#L2), for example)

@wjwwood and I also considered using regex subroutines or having capture groups both before and after the regex and checking that they're empty, but subroutines don't seem to be supported in python for the former approach, and for the latter the same situation is introduced with the group numbering being changed.